blog/_posts/2018-06-13-scala-213-collections.md

```diff
@@ -0,0 +1,210 @@
+their elements in place: instead of returning a new collection (like
+`map` and `filter` do) they mutate the source collection. These
+operations are suffixed with `InPlace`. For instance, to remove
+users whose name start with the letter `J` from a buffer and then
+increment their age, one can now write:
+
+~~~ scala
+val users = ArrayBuffer(…)
+users
+  .filterInPlace(user => !user.name.startsWith("J"))
+  .mapInPlace(user => user.copy(age = user.age + 1))
+~~~
+
+## Deprecations For Less Confusion
+
+A consequence of cleaning and simplifying the collections framework
+is that several types or operations have been deprecated.
+
+### `Iterable` Is The Top Collection Type
+
+We felt that having a distinction between `Traversable` and `Iterable` was not
+worth it, so we removed `Traversable` (it is now an alias to `Iterable[A]`).
+
+`IterableOnce[A]` is now the collection type at the top of the hierarchy.
+Its only abstract member is `def iterator: Iterator[A]`.
+
+### `LazyList` Is Preferred Over `Stream`
+
+`Stream` is deprecated in favor of `LazyList`. As its name suggests,
+a `LazyList` is a linked list whose elements are lazily evaluated. An
+important semantic difference with `Stream` is that in `LazyList` both
+the head and the tail are lazy, whereas in `Stream` only the tail is lazy.
+
+### Insertion And Removal Operations Are Not Available On Generic Collections
+
+In the current framework, the `scala.collection.Map` type has a `+` and a `-` operations
+to add and remove entries. The semantics of these operations is to return a new collection
+with the added or removed entries, without changing the source collection.
+
+These operations are then inherited by the mutable branch of the collections. But the mutable
+collection types also introduce their own insertion and removal operations, namely `+=` and `-=`,
+which modify the source collection in place. This means that the `scala.collection.mutable.Map` type
+has `+` and `+=`, as well as `-` and `-=`.
+
+Having all these operations can be handy in some cases but can also introduce confusion. If you want
+to use `+` or `-`, then you probably wanted to use an immutable collection type in the first place…
+Another example is the `updated` operation, which is available on mutable `Map` but returns a new
+collection.
+
+We think that by deprecating these insertion and removal operations from generic collection
+types and by having distinct operations between the `mutable` and `immutable` branches we make
+the situation clearer.
+ 
+## Summary
+
+In summary, the changes for end-users are the following:
+
+- non-strict collections (such as views) are safer to use and simpler implement,
+- type signatures of transformation operations (such as `map`) are simpler
+  (no implicit `CanBuildFrom` parameter),
+- new cool operations have been added,
+- the type hierarchy is simpler (no `Traversable`),
+- mutable collection types do not inherit immutable insertion and removal operations.
```

Comment:
Again, be more precise about in which version of Scala which specific methods will be deprecated, and in which version of Scala they will be removed.